File: Python/70-Climbing_Stairs.py
class Solution(object):
  def climbStairs(self, n):
    # O(n) time | O(1) space
    table = [1, 1]
    
    for i in range(2, n + 1):
        temp = table[0]
        table[0] = table[1]
        table[1] += temp
    return table[1]

File: Javascript/70-Climbing_Stairs.js
var climbStairs = function (n) {
  // O(n) time | O(1) space
  const pair = [1, 1];

  for (let i = 2; i <= n; i++) {
    const temp = pair[0];
    pair[0] = pair[1];
    pair[1] += temp;
  }

  return pair[1];
};
